lib/flatfile/QubitFlatfileExport.class.php

```diff
@@ -296,24 +308,82 @@ public function prepareRowFromResource()
         foreach ($this->columnNames as $index => $column) {
             $value = $this->row[$index];
 
-            // If row value hasn't been set to anything, attempt to get resource property
-            if (null === $value) {
+            // If row value hasn't been set to anything and element is not hidden,
+            // attempt to get resource property
+            if (null === $value && !in_array($column, $this->nonVisibleElementsIncluded)) {
                 if (in_array($column, $this->standardColumns)) {
                     $value = $this->resource->{$column};
                 } elseif (($sourceColumn = array_search($column, $this->columnMap)) !== false) {
                     $value = $this->resource->{$sourceColumn};
                 } elseif (isset($this->propertyMap[$column])) {
                     $value = $this->resource->getPropertyByName($this->propertyMap[$column])->__toString();
                 }
-            }
 
-            // Add column value (imploding if necessary)
-            $this->row[$index] = $this->content($value);
+                // Add column value (imploding if necessary)
+                $this->row[$index] = $this->content($value);
+            } else {
+                // Unset hidden elements columns
+                unset($this->row[$index]);
+            }
         }
 
         $this->modifyRowBeforeExport();
     }
 
+    /**
+     * Get parameters to determine hidden elements.
+     *
+     * @param array parameters for CSV export
+     * @param mixed $params
+     */
+    public function passParams($params)
```

Comment:
I would suggest naming this function `setParams` since this behaves as a setter function